media-sound/flacon/flacon-8.2.0.ebuild

```diff
@@ -0,0 +1,83 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+# Tests require lots of disk space
+CHECKREQS_DISK_BUILD=10G
+inherit check-reqs cmake optfeature virtualx xdg-utils
+
+DESCRIPTION="Extracts audio tracks from an audio CD image to separate tracks"
+HOMEPAGE="https://flacon.github.io/"
+SRC_URI="https://github.com/flacon/flacon/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="LGPL-2.1+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+
+BDEPEND="
+	virtual/pkgconfig
+	dev-qt/linguist-tools:5
+	media-libs/taglib
+"
+RDEPEND="
+	app-i18n/uchardet
+	dev-qt/qtcore:5
+	dev-qt/qtgui:5
+	dev-qt/qtnetwork:5
+	dev-qt/qtwidgets:5
+"
+DEPEND="${RDEPEND}
+	dev-qt/qtconcurrent:5
+	test? (
+		dev-qt/qttest:5
+		media-libs/flac
+		media-sound/mac
+		media-sound/shntool
+		media-sound/ttaenc
+		media-sound/wavpack
+	)
+"
+
+RESTRICT="!test? ( test )"
+
+pkg_pretend() {
+	use test && check-reqs_pkg_pretend
+}
+
+pkg_setup() {
+	use test && check-reqs_pkg_setup
+}
+
+src_configure() {
+	local mycmakeargs=(
+		-DBUILD_TESTS="$(usex test)"
+	)
+	cmake_src_configure
+}
+
+src_test() {
+	virtx "${BUILD_DIR}/tests/${PN}_test"
```

Comment:
Careful!

```
# @FUNCTION: virtx
# @USAGE: <command> [command arguments]
# @DESCRIPTION:
# Start new Xvfb session and run commands in it.
#
# IMPORTANT: The command is run nonfatal !!!
#
# This means we are checking for the return code and raise an exception if it
# isn't 0. So you need to make sure that all commands return a proper
# code and not just die. All eclass function used should support nonfatal
# calls properly.
```

Reply:
To be honest.  The tests fail anyway on the new and the old version. Don't know why but sometimes they pass and sometimes they don't.

Reply:
Still, we should fix the usage, but you can restrict the tests if they're inconsistent.

Reply:
Yes you're right just wanted to inform you. 
Thanks for the feedback btw.